src/scripts/run_certbot.sh

```diff
@@ -87,6 +114,14 @@ for conf_file in /etc/nginx/conf.d/*.conf*; do
             key_type="rsa"
         fi
 
+        # Determine the authenticator to use to solve the challenge
+        if [[ "${cert_name,,}" =~ (^|[^a-z0-9])(dns-($(echo ${CERTBOT_DNS_AUTHENTICATORS} | sed 's/ /|/g')))([^a-z0-9]|$) ]]; then
```

Comment:
I am not sure changing to `(^|[^a-z0-9])` and `([^a-z0-9]|$)` is such a good idea, I think we should keep the delimiter consistent between this and the `key_type` lookup and `^a-z0-9` allows way too many options in my opinion :)

Reply:
You're right, I can keep the same delimiters, this is too broad.
However, I'll keep the `(^|[.-])` and  `([.-]|$)` if you're good with that, as they allow for delimited-only (either by one of the delimiters or the beginning/end of the string) matching, vs `dns-cloudflarey` matching because it starts by `dns-cloudflare`. What do you think ?
I believe we could apply the same matching to RCC/RSA also, as it's backward compatible too.

Reply:
That is really clean, I like it a lot!
Sure, this should be copied to the key matching, but let's do that after this is merged :)